src/main/java/graphql/execution/preparsed/persisted/PersistedQueryCache.java

```diff
@@ -27,5 +29,5 @@
      * @return a parsed and validated PreparsedDocumentEntry where {@link graphql.execution.preparsed.PreparsedDocumentEntry#getDocument()} is set
      * @throws graphql.execution.preparsed.persisted.PersistedQueryNotFound if the query id is not know at all and you have no query text
      */
-    PreparsedDocumentEntry getPersistedQueryDocument(Object persistedQueryId, ExecutionInput executionInput, PersistedQueryCacheMiss onCacheMiss) throws PersistedQueryNotFound;
+    CompletableFuture<PreparsedDocumentEntry> getPersistedQueryDocument(Object persistedQueryId, ExecutionInput executionInput, PersistedQueryCacheMiss onCacheMiss) throws PersistedQueryNotFound;
```

Comment:
This is indeed a great idea.   This should indeed be an async method because its likely to go to a DB etc...for most implementations.


However as implemented this is a hard breaking change.

We can avoid this in this case.

We can

* deprecate the `PreparsedDocumentEntry getPersistedQueryDocument(` method
* put in a  `default CompletableFuture<PreparsedDocumentEntry> getPersistedQueryDocumentAsync(` method
* make the default call the `getPersistedQueryDocument` and wrap it in a completed CF
* make all the calls sites called `CompletableFuture<PreparsedDocumentEntry> getPersistedQueryDocumentAsync(` and hence it will invoke the sync version for every one else.